packages/std/src/storage.rs

```diff
@@ -31,13 +33,25 @@ impl ReadonlyStorage for MemoryStorage {
         start: Option<&[u8]>,
         end: Option<&[u8]>,
         order: Order,
-    ) -> Box<dyn Iterator<Item = KV> + 'a> {
+    ) -> Result<Box<dyn Iterator<Item = KV> + 'a>> {
         let bounds = range_bounds(start, end);
+
+        // BTreeMap.range panics if range is start > end.
+        // However, this cases represent just empty range and we treat it as such.
```

Comment:
panic is bad.

Consider turning an Error here, to help devs see this is incorrect usage (often by mistake). Returning empty may confuse them when they expect this to return data. But this is open to debate.

Reply:
Ok